### DISCUSSION TITLE
Call a scoped service method inside OnReceived #198

### DISCUSSION THREAD

**Discussion by samnemo94** [jchristn/SuperSimpleTcp#140] [General]
Consider the "Simple Server Example"; inside DataReceived method: I want to call an async method of a scoped service.
So I have two questions:
1- How can I pass the scoped service to the DataReceived? I think it is not recommended to put it as static variable and access it statically?
2- How can I make the DataReceived method async because the method I want to invoke is async?
Use Case:
I'm implementing two micro-services, one is the socket server that listens to a port and invokes RPC method from another service so the RPC invocation requires async.

**Comment by jchristn** [comment]
Hi @samnemo94 the library doesn't provide a facility to pass such methods into the data receiver loop.  However, you are welcome to clone, modify, and use the source code as you see fit - it's MIT licensed, meaning you can really do with it as you wish.  If there is a capability that fits cleanly into the current library architecture, I'd love a PR on such a capability.  Moving this into a discussion.  Cheers, Joel
